### ISSUE TITLE
aruco: build warnings (MSVS 2015)

### ISSUE THREAD

**Issue by alalek** [opencv/opencv_contrib#3327]
relates #3240

Logs: http://pullrequest.opencv.org/buildbot/builders/precommit_windows64/builds/100075
Logs (2): http://pullrequest.opencv.org/buildbot/builders/master-contrib-win64-vc14/builds/100000

> C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\memory(901): warning C4244: 'argument': conversion from 'const int' to 'float', possible loss of data (compiling source file C:\build\precommit_windows64\4.x\opencv_contrib\modules\aruco\src\charuco.cpp) [C:\build\precommit_windows64\build\modules\aruco\opencv_aruco.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\memory(901): warning C4244: 'argument': conversion from 'const double' to 'float', possible loss of data (compiling source file C:\build\precommit_windows64\4.x\opencv_contrib\modules\aruco\test\test_boarddetection.cpp) [C:\build\precommit_windows64\build\modules\aruco\opencv_test_aruco.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\memory(901): warning C4244: 'argument': conversion from 'const int' to 'float', possible loss of data (compiling source file C:\build\precommit_windows64\4.x\opencv_contrib\modules\aruco\test\test_boarddetection.cpp) [C:\build\precommit_windows64\build\modules\aruco\opencv_test_aruco.vcxproj]

Check full compiler log for details.

**Comment by alalek** [comment]
Bug, found by GCC 12.1 (Fedora 36):

```
[3032/3540] Building CXX object modules/aruco/CMakeFiles/opencv_aruco.dir/src/aruco_detector.cpp.o
/home/alalek/projects/opencv/contrib/modules/aruco/src/aruco_detector.cpp: In member function ‘bool cv::aruco::DetectorParameters::writeDetectorParameters(const cv::Ptr<cv::FileStorage>&)’:
/home/alalek/projects/opencv/contrib/modules/aruco/src/aruco_detector.cpp:66:36: warning: ‘this’ pointer is null [-Wnonnull]
   66 |     if (fs.empty() && !fs->isOpened())
      |                        ~~~~~~~~~~~~^~
```

**Comment by AleksandrPanov** [comment]
@alalek, fixed bug by #3346
I use gcc 9.3.0 and flags `-Wconversion`/`-Wfloat-conversion` to find lines of code with warnings. But gcc can't find warning "C4244: 'argument': conversion from 'const int' to 'float', possible loss of data"

Could you advise the flags for GCC to find these warnings?

Or show lines that have problems?
